ctapipe/irf/irf_classes.py

```diff
@@ -0,0 +1,74 @@
+"""
+Define a parent IrfTool class to hold all the options
+"""
+from enum import Enum
+
+import astropy.units as u
+from pyirf.cuts import calculate_percentile_cut
+from pyirf.spectral import CRAB_HEGRA, IRFDOC_ELECTRON_SPECTRUM, IRFDOC_PROTON_SPECTRUM
+
+from ..core import Component
+from ..core.traits import Float, Integer
+
+
+class Spectra(Enum):
+    CRAB_HEGRA = 1
+    IRFDOC_ELECTRON_SPECTRUM = 2
+    IRFDOC_PROTON_SPECTRUM = 3
+
+
+PYIRF_SPECTRA = {
+    Spectra.CRAB_HEGRA: CRAB_HEGRA,
+    Spectra.IRFDOC_ELECTRON_SPECTRUM: IRFDOC_ELECTRON_SPECTRUM,
+    Spectra.IRFDOC_PROTON_SPECTRUM: IRFDOC_PROTON_SPECTRUM,
+}
+
+
+class ThetaCutsCalculator(Component):
+    theta_min_angle = Float(
+        default_value=-1, help="Smallest angular cut value allowed (-1 means no cut)"
+    ).tag(config=True)
+
+    theta_max_angle = Float(
+        default_value=0.32, help="Largest angular cut value allowed"
+    ).tag(config=True)
+
+    theta_min_counts = Integer(
+        default_value=10,
+        help="Minimum number of events in a bin to attempt to find a cut value",
+    ).tag(config=True)
+
+    theta_fill_value = Float(
+        default_value=0.32, help="Angular cut value used for bins with too few events"
+    ).tag(config=True)
+
+    theta_smoothing = Float(
+        default_value=-1,
+        help="When given, the width (in units of bins) of gaussian smoothing applied (-1)",
+    ).tag(config=True)
+
+    target_percentile = Float(
+        default_value=68,
+        help="Percent of events in each energy bin keep after the theta cut",
+    ).tag(config=True)
+
+    def calculate_theta_cuts(self, theta, reco_energy, energy_bins):
```

Comment:
Classes that basically are configurable functions (i.e. have traitlets and then a single method) should use `__call__` I think.

Reply:
After recently having tried to explain how `__call__` means you can do object(args) to a beginner, I think that syntax has too much magic and explicit is better.

Reply:
I think it's more important to be consistent. If that's a decision we should make, we should make it everywhere and not just here. However, regarding this particular point, I have a different opinion.

We have many instances in ctapipe now that implement `__call__` and I think it's the correct thing to do. 

Beginners have problems with many concepts, that shouldn't mean that we optimize our code for being understandable by beginners vs. writing idiomatic python.